wallet-new/src/Cardano/Wallet/WalletLayer/Legacy.hs

```diff
@@ -159,24 +181,24 @@ pwlUpdateWallet wId wUpdate = do
     setWalletMeta walletDB cWId cWMeta
 
     -- Get wallet or throw if missing.
-    maybeThrow (WalletNotFound wId) =<< pwlGetWallet wId
+    pwlGetWallet wId
```

Comment:
If `pwlGetWallet` returns `Maybe`, then this logic also needs to change.

If we update a wallet that doesn't exist, what happens? Seems like the same problem: it should return a value indicating `WalletNotFound` (whether that is `Nothing` or a `Left (WalletNotFound ...)`. Let's reflect that in the types.